### PR TITLE
Increase code coverage of Context

### PR DESCRIPTION
**Pull request by ejona86**
A few minor bugs were discovered and fixed.

### CODE REVIEW

File: core/src/main/java/io/grpc/Context.java [hunk 1]
```diff
@@ -759,7 +755,11 @@ private ExecutableListener(Executor executor, CancellationListener listener) {
     }
 
     private void deliver() {
-      executor.execute(this);
+      try {
+        executor.execute(this);
+      } catch (Throwable t) {
+        LOG.log(Level.INFO, "Exception notifying context listener", t);
```
Comment (carl-mastrangelo) [inline core/src/main/java/io/grpc/Context.java]: This catches Interrupted.  Is this okay?

Comment (ejona86) [inline core/src/main/java/io/grpc/Context.java]: `Executor.execute()` does not throw `InterruptedException`, which as a checked exception.

File: core/src/test/java/io/grpc/ContextTest.java [hunk 2]
```diff
@@ -407,4 +644,61 @@ public void innerDeadlineCompletesBeforeOuter() throws Exception {
     assertTrue(base.cause() instanceof TimeoutException);
     assertNotSame(base.cause(), child.cause());
   }
+
+  @Test
+  public void cancellationCancelsScheduledTask() {
+    ScheduledExecutorService service = SharedResourceHolder.get(Context.SCHEDULER);
+    try {
+      ThreadPoolExecutor executor = (ThreadPoolExecutor) service;
+      //executor.setRemoveOnCancelPolicy(true);
```
Comment (carl-mastrangelo) [inline core/src/test/java/io/grpc/ContextTest.java]: remove comment?

File: core/src/test/java/io/grpc/ContextTest.java [hunk 3]
```diff
@@ -407,4 +644,61 @@ public void innerDeadlineCompletesBeforeOuter() throws Exception {
...
+      final CountDownLatch ran = new CountDownLatch(1);
+      service.execute(new Runnable() {
+        @Override
+        public void run() {
+          threadName.set(Thread.currentThread().getName());
+          threadIsDaemon.set(Thread.currentThread().isDaemon());
+          ran.countDown();
+        }
+      });
+      ran.await();
```
Comment (carl-mastrangelo) [inline core/src/test/java/io/grpc/ContextTest.java]: any reason to not just get the future from service.execute and call .get() on that?

File: core/src/test/java/io/grpc/ContextTest.java (hunk 2 above)
Comment (ejona86) [inline core/src/test/java/io/grpc/ContextTest.java]: Done. It was done not quite 30 minutes ago. If you refresh you should see it as gone.

**Comment by carl-mastrangelo** [comment]
@ejona86  A few minor nits, but LGTM

File: core/src/test/java/io/grpc/ContextTest.java (hunk 3 above)
Comment (ejona86) [inline core/src/test/java/io/grpc/ContextTest.java]: Nope. I think I just got into the habit of not having `submit()` available. Done; now using `submit()` and calling `Future.get()`.

**Pull request merged**